Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -895,6 +895,7 @@ internal struct BinaryDecoder: Decoder {
 
         if let k = k, let v = v {
             value[k] = v
+            consumed = true
```

Comment:
I believe your proposed fix here is correct.

I'm a little dismayed that our tests did not catch this earlier; I should go back and figure out what tests we're missing.

As a minor matter of style, I would put this statement outside of this conditional, and might even pull it up to just after `getFieldBodyBytes` above.  When you put it here, it makes `consumed` look like it's flagging decoding success when it's really intended to just mark that the field has been inspected.